tc/tc_config.h.in

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#define TC_STRINGIFY_INNER(s) #s
+#define TC_STRINGIFY(s) TC_STRINGIFY_INNER(s)
+
+// clang-format off
+#define TC_DIR @TC_DIR@
+#define TC_WITH_CUDA @TC_WITH_CUDA@
+#define TC_CUDA_TOOLKIT_ROOT_DIR @TC_CUDA_TOOLKIT_ROOT_DIR@
+#define TC_CUDA_INCLUDE_DIR @TC_CUDA_INCLUDE_DIR@
+#define TC_CUDA_LIBRARIES @TC_CUDA_LIBRARIES@
+#define TC_CUB_INCLUDE_DIR @TC_CUB_INCLUDE_DIR@
```

Comment:
Why not simply export these as, e.g., `"@TC_CUB_INCLUDE_DIR@"` instead of stringifying them afterwards?

Reply:
> because it would create issues with `#if defined` type logic

What kind of issues?

skimo

Reply:
When a macro is defined to `""` it is not trivial to detect

Reply:
> When a macro is defined to `""` it is not trivial to detect

I don't get it.
What's the difference between A and B in

```
#define A
#define B ""
```
?

`defined(A)` and `defined(B)` both evaluate to true.

skimo

Reply:
Btw, you can easily check for an empty string using `static_assert` and `sizeof`.